src/combinators.js

```diff
@@ -90,3 +90,25 @@ export function guardDepth(fuzzerA, fuzzerB) {
     fuzzerState.tooDeep() ? fuzzerA(fuzzerState) : fuzzerB(fuzzerState);
 }
 
+export function guardDuplicatedProto(fuzzer) {
+  return (fuzzerState = new FuzzerState) => {
+    let fuzzed = many(fuzzer)(fuzzerState);
```

Comment:
You should probably pass the combinator (`many`) in. Then in the tests you can use `manyN` to ensure to arbitrary certainty that this does the right thing.